src/ocean/text/Util.d

```diff
@@ -785,6 +788,136 @@ bool isSpace(T) (T c)
            return (c <= 32 && (c is ' ' || c is '\t' || c is '\r' || c is '\n' || c is '\f' || c is '\v')) || (c is '\u2028' || c is '\u2029');
 }
 
+/*******************************************************************************
+
+    Enumeration of the different ways in which a string can be padded in order
+    to increase its length.
+
+*******************************************************************************/
+
+public enum PadType : ubyte
+{
+    PadPre,    // Padding is only applied before the string
+    PadPost,   // Padding is only applied after the string
+    PadAround  // Padding is applied on both sides of the string
+}
+
+/*******************************************************************************
+
+    Shortens/Lengthens a string as necessary to get it to a desired length.
+
+    Note 1:
+        If the string is shortened, an ellipsis (3 dots) will be appended to
+        indicate intentional truncation.
+
+    Note 2:
+        The length specified must be at least 4. If this constraint were absent
+        and a length of 3 or lower was specified, then the input string would be
+        converted simply to an ellipsis, losing all information about the
+        string.
+
+    Note 3:
+        If the input string needs to be lengthened and padding on both sides
```

Comment:
>I should add here that this function is not really new. It's been running in some apps (closed-source unfortunately) since as far back as July 2014.

By the way, it's typical that when code is moved from a project to a library, its structure is reassessed. The requirements of library code are (obviously) quiet different.